apollo-runtime/src/main/java/com/apollographql/apollo/internal/RealApolloCall.java

```diff
@@ -68,9 +72,8 @@
   final List<Query> refetchQueries;
   final Optional<QueryReFetcher> queryReFetcher;
   final boolean sendOperationdIdentifiers;
-  final AtomicBoolean executed = new AtomicBoolean();
-  volatile boolean canceled;
-  volatile boolean completed;
+  final AtomicReference<CallState> state = new AtomicReference<>(IDLE);
```

Comment:
I believe we can omit `AtomicReference ` here as we don't use any CAS operations, `volatile` should be enough. The same for callback